apps/api/src/app/execution-details/execution-details.controller.ts

```diff
@@ -36,4 +46,27 @@ export class ExecutionDetailsController {
       })
     );
   }
+
+  @Get('/transaction/:transactionId')
+  @ApiResponse(ExecutionDetailsFilterResponseDto)
+  @ApiOperation({
+    summary: 'Get execution details by transaction id',
+  })
+  @ExternalApiAccessible()
+  getNotificationTemplates(
+    @UserSession() user: IJwtPayload,
+    @Param('transactionId') transactionId: string,
+    @Query() query: GetExecutionDetailsByTransactionRequestDto
+  ): Promise<ExecutionDetailsFilterResponseDto> {
+    return this.getExecutionDetailsByTransactionId.execute(
+      GetExecutionDetailsByTransactionIdCommand.create({
+        transactionId,
+        organizationId: user.organizationId,
+        userId: user._id,
+        environmentId: user.environmentId,
+        page: query.page ? query.page : 0,
+        limit: query.limit ? query.limit : 1,
```

Comment:
```suggestion
        limit: query.limit ? query.limit : 10,
```
I feel 10 is a reasonable default limit